### PR TITLE
feat(drawer): typescript support 

### PR DESCRIPTION
**Pull request by moog16**
related #48

### CODE REVIEW

File: packages/drawer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        const hasPreviousFocus =
+          this.previousFocus_ && this.previousFocus_.focus;
+        const drawerElement =
+          this.drawerElement_ && this.drawerElement_.current;
+        if (
+          drawerElement &&
+          hasPreviousFocus &&
+          drawerElement.contains(document.activeElement)
+        ) {
+          this.previousFocus_!.focus();
```
Comment (rlfriedman) [inline packages/drawer/index.tsx]: Is the ! here intentional? Formatting of this if statement also seems a bit weird.

File: packages/drawer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        }
+      },
+      focusActiveNavigationItem: () => {
+        const drawerElement =
+          this.drawerElement_ && this.drawerElement_.current;
+        if (!drawerElement) return;
+        const activeNavItemEl = drawerElement.querySelector(
+          `.${listCssClasses.LIST_ITEM_ACTIVATED_CLASS}`
+        ) as HTMLElement | null;
```
Comment (rlfriedman) [inline packages/drawer/index.tsx]: Formatting/lack of indentation here is a bit confusing.

File: packages/drawer/index.tsx (hunk 1 above)
Comment (moog16) [reply packages/drawer/index.tsx]: UGH! it was the formatter...
And yes is intentional. For some reason the tslint doesn't see the IF check I have for this.previousFocus

**Comment by moog16** [comment]
@rlfriedman updated!

**Pull request merged**